libs/vkd3d/device.c

```diff
@@ -4701,10 +4705,20 @@ static void d3d12_device_caps_init_feature_options6(struct d3d12_device *device)
 
     options6->AdditionalShadingRatesSupported = device->device_info.additional_shading_rates_supported;
     options6->VariableShadingRateTier = d3d12_device_determine_variable_shading_rate_tier(device);
+    if (options6->VariableShadingRateTier >= D3D12_VARIABLE_SHADING_RATE_TIER_2)
+    {
+        options6->ShadingRateImageTileSize = max(
```

Comment:
This is not quite correct. D3D12 requires us to expose either 8x8, 16x16 or 32x32 statically, so the ideal thing to do is to query 8x8, 16x16, 32x32 and select the first one we find that is supported.

Reply:
There are no vendors exposing a different min granularity than max granularity right now -- so, don't support this if it isn't in that set?

Reply:
Yes, if the driver cannot satisfy one of the required tile sizes, don't expose TIER_2.